src/components/safe-apps/AppFrame/index.tsx

```diff
@@ -0,0 +1,76 @@
+import { ReactElement, useCallback, useEffect } from 'react'
+import { CircularProgress, Typography } from '@mui/material'
+import { trackSafeAppOpenCount } from '@/services/safe-apps/track-app-usage-count'
+import { useSafeAppFromManifest } from '@/hooks/safe-apps/useSafeAppFromManifest'
+import { useSafeApps } from '@/hooks/safe-apps/useSafeApps'
+import useSafeInfo from '@/hooks/useSafeInfo'
+import { isSameUrl } from '@/utils/url'
+import { ThirdPartyCookiesWarning } from './ThirdPartyCookiesWarning'
+import useThirdPartyCookies from './useThirdPartyCookies'
+import useAppIsLoading from './useAppIsLoading'
+
+import css from './styles.module.css'
+
+export type TransactionParams = {
+  safeTxGas?: number
+}
+
+type AppFrameProps = {
+  appUrl: string
+}
+
+const AppFrame = ({ appUrl }: AppFrameProps): ReactElement => {
+  const { safe } = useSafeInfo()
+  const { remoteSafeApps } = useSafeApps()
+  const { safeApp: safeAppFromManifest } = useSafeAppFromManifest(appUrl, safe.chainId)
+  const { thirdPartyCookiesDisabled, setThirdPartyCookiesDisabled } = useThirdPartyCookies()
+  const { iframeRef, appIsLoading, isLoadingSlow, setAppIsLoading } = useAppIsLoading()
+
+  const currentApp = remoteSafeApps?.filter((app: { url: string }) => app.url === appUrl)[0]
```

Comment:
I know it's copied from safe-react, but I would put this into the useEffect bellow instead. It's not used anywhere else in the component.

Reply:
added a useMemo. Is going to be used  for the Transaction Modal in later steps. The component is going to grow when adding all the sdk and communication stuff